mslib/msui/mpl_pathinteractor.py

```diff
@@ -711,36 +1062,9 @@ def button_release_insert_callback(self, event):
         self._ind = None
 
     def get_lat_lon(self, event):
-        x = event.xdata
         wpm = self.waypoints_model
-        vertices = self.pathpatch.get_path().vertices
-        best_index = 1
-        # if x axis has increasing coordinates
-        if vertices[-1, 0] > vertices[0, 0]:
-            for index, vertex in enumerate(vertices):
-                if x >= vertex[0]:
-                    best_index = index + 1
-        # if x axis has decreasing coordinates
-        else:
-            for index, vertex in enumerate(vertices):
-                if x <= vertex[0]:
-                    best_index = index + 1
-        # number of subcoordinates is determined by difference in x coordinates
-        number_of_intermediate_points = math.floor(vertices[best_index, 0] - vertices[best_index - 1, 0])
-        vert_xs, vert_ys = latlon_points(
-            vertices[best_index - 1, 0], vertices[best_index - 1, 1],
-            vertices[best_index, 0], vertices[best_index, 1],
-            number_of_intermediate_points, connection="linear")
-        lats, lons = latlon_points(
-            wpm.waypoint_data(best_index - 1).lat, wpm.waypoint_data(best_index - 1).lon,
-            wpm.waypoint_data(best_index).lat, wpm.waypoint_data(best_index).lon,
-            number_of_intermediate_points, connection="greatcircle")
-
-        # best_index1 is the best index among the intermediate coordinates to fit the hovered point
-        # if x axis has increasing coordinates
-        best_index1 = np.argmin(abs(vert_xs - x))
-        # depends if best_index1 or best_index1 - 1 on closeness to left or right neighbourhood
-        return (lats[best_index1], lons[best_index1]), best_index
+        lat_lon, ind = self.plotter.get_lat_lon(event, wpm)
```

Comment:
I would prefer, for consistency, when you do not supply the wpm model, but only the contained data (similar to redraw_path).
Think about testing! For testing this get_lat_lon function, one needs to create/mock a complicated QT structure, where a simple list of coordinates'd work as well.